framework/src/modules/chain/blocks/blocks.js

```diff
@@ -493,6 +493,24 @@ class Blocks extends EventEmitter {
 		}
 	}
 
+	/**
+	 * Returns the largest common block between ids.ids and the database blocks table
+	 * @param {Array<String>} ids - An array of block ids
+	 * @return {Promise<BasicBlock|undefined>}
+	 */
+	async getHighestCommonBlock(ids) {
+		try {
+			const block = await this.storage.entities.Block.getMatchingHighestBlock(
+				ids
+			);
+			return block;
+		} catch (e) {
+			const errMessage = 'Failed to read common blocks from storage';
```

Comment:
Failed to read common blocks implies that you couldn't find a common block.
However, an error is only thrown in case of bad network connection or unable to access storage layer.
Prefer to throw more generic message like `Failed to access storage layer/or network`.